salt/states/zabbix_user.py

```diff
@@ -23,7 +24,101 @@ def __virtual__():
     return 'zabbix.user_create' in __salt__
 
 
-def present(alias, passwd, usrgrps, medias, password_reset=False, **kwargs):
+def admin_password_present(name, password=None, **kwargs):
+    '''
+    Initial change of Zabbix Admin password to password taken from one of the sources (only the most prioritized one):
+        1. 'password' parameter
+        2. '_connection_password' parameter
+        3. pillar 'zabbix.password' setting
+
+    1) Tries to log in as Admin with password found in state password parameter or _connection_password
+       or pillar or default zabbix password in this precise order order, if any of them is present.
+    2) If one of above passwords found is successful it tries to update the most prioritized password.
+    3) If not able to connect with any password then it fails.
+
+    :param name: Just a name of state
+    :param password: Optional - desired password for Admin to be set
+    :param _connection_user: Optional - Ignored in this state (always assumed 'Admin')
+    :param _connection_password: Optional - zabbix password (can also be set in opts or pillar, see module's docstring)
+    :param _connection_url: Optional - url of zabbix frontend (can also be set in opts, pillar, see module's docstring)
+
+    .. code-block:: yaml
+
+    # password taken from pillar or _connection_password
```

Comment:
This whole block to the end of the example needs to be indented for the code-block to render correctly.